scripts/rpc-maas-tool.py

```diff
@@ -292,13 +309,199 @@ def main(self):
+            """Execute one check
+
+            This function will be called from a threadpool thread.
+            """
+            try:
+                result = {'success': True,
+                          'output': subprocess.check_output(
+                              args,
+                              stderr=subprocess.STDOUT),
+                          'check': check
+                          }
+            except subprocess.CalledProcessError as e:
+                result = {'success': False,
+                          'output': e.output,
+                          'check': check
+                          }
+            rpcm.q.put(result)
+
+        # Checks are executed using a threadpool to speed up verification.
+        execution_results = []
+        with concurrent.futures.ThreadPoolExecutor(max_workers=10) as executor:
+            for check in self.rpcmac.checks.values():
+                if check['type'] != 'agent.plugin':
+                    continue
+                args = ["{plugin_path}{plugin}".format(
+                    plugin_path='/usr/lib/rackspace-monitoring-agent/plugins/',
+                    plugin=check['details']['file'])]
+                args.extend(check['details']['args'])
+                executor.submit(_execute_check, args, check, self.rpcm)
+        while not self.rpcm.q.empty():
+            execution_results.append(self.rpcm.q.get())
+
+        available_metrics = set()  # metrics produced by check commands
+        required_metrics = set()  # metrics used in alarm criteria
+        failed_checks = []  # checks that failed to execute or not 'okay'
+        invalid_criteria = []  # alarms with invalid criteria
+
+        for result in execution_results:
+            check = result['check']
+            if result['success'] is not True:
+                failed_checks.append(result)
+                continue
+            # use the output of executing the checks to find available metrics
+            for line in result['output'].splitlines():
+                # check the status line and return code
+                match = status_line_re.match(line)
+                if match:
+                    status = match.groupdict()['status']
+                    if status != 'okay':
+                        failed_checks.append(result)
+                # store all the metrics that are returned by the check
+                match = metric_line_re.match(line)
+                if match:
+                    available_metrics.add(match.groupdict()['name'])
+
+        # Parse alarm criteria
+        for check in self.rpcmac.checks.values():
+            try:
+                metrics = _alarm_metrics_from_check(check)
+                # Non agent metrics are not added to required_metrics because
+                # we can't determine locally the metrics that will be
+                # available to alarms for remote checks.
+                if check['type'] == 'agent.plugin':
```

Comment:
I can only execute agent.plugin checks, so I only know the available metrics for agent.plugin checks. Other checks have their alarm criteria syntax checked, but metric names are not checked. How would I determine the metrics produced by an agent.filesystem check? 

My comment on L456 should say "non agent.plugin" metrics, rather than non-agent.